extensions/wikia/AdEngine/js/provider/directGpt.js

```diff
@@ -54,30 +58,88 @@ define('ext.wikia.adEngine.provider.directGpt', [
 			CORP_TOP_RIGHT_BOXAD: 'flush',
 			TOP_RIGHT_BOXAD:      'flush',
 			HOME_TOP_RIGHT_BOXAD: 'flush',
-			GPT_FLUSH:            'flushonly'
-		};
-
-	return factory.createProvider(
-		logGroup,
-		'DirectGpt',
-		'gpt',
-		slotMap,
-		{
-			beforeSuccess: function (slotName) {
-				slotTweaker.removeDefaultHeight(slotName);
-				slotTweaker.removeTopButtonIfNeeded(slotName);
-				slotTweaker.adjustLeaderboardSize(slotName);
-			},
-			shouldFlush: function (slotName) {
-				log(['shouldFlush', slotName]);
+			GPT_FLUSH:            'flush'
+		},
+		provider = factory.createProvider(
+			logGroup,
+			'DirectGpt',
+			'gpt',
+			slotMap,
+			{
+				beforeSuccess: function (slotName) {
+					slotTweaker.removeDefaultHeight(slotName);
+					slotTweaker.removeTopButtonIfNeeded(slotName);
+					slotTweaker.adjustLeaderboardSize(slotName);
+					removePendingSlot(slotName);
+				},
+				beforeHop: function (slotName) {
+					removePendingSlot(slotName);
+				},
+				shouldFlush: function (slotName) {
+					log(['shouldFlush', slotName]);
+					if (gptConfig[slotName] === 'flush') {
+						gptFlushed = true;
+					}
 
-				if (gptConfig[slotName] === 'flushonly' || gptConfig[slotName] === 'flush') {
-					gptFlushed = true; // Setting the module-scope var here
+					log(['shouldFlush', slotName, gptFlushed]);
+					return gptFlushed;
 				}
+			}
+		);
 
-				log(['shouldFlush', slotName, gptFlushed]);
-				return gptFlushed;
+	function removePendingSlot(slotName) {
```

Comment:
I'd suggest renaming this function or removing the condition connected to pushing delayed queue and placing it within a new function :japanese_ogre: